lib/procore/requestable.rb

```diff
@@ -297,7 +297,11 @@ def multipart?(body)
     end
 
     def full_path(path)
-      File.join(base_api_path, path).to_s
+      File.join(rest_compatible_base_api_path(path), path).to_s
```

Comment:
It might make more sense to change the definition of `base_api_path`.  We could accept a separate option for `api_version` (or a different name), which defaults to `vapid`.  `base_api_path` then becomes:
```ruby
def base_api_path
  "#{options[:host]}/#{options[:api_version]}"
end
```
This solution would require a separate `Procore::Client` per API version, which IMO is acceptable.  Requiring the client-side implementation to be explicit lends to a more predictable experience.

Reply:
I can do that.

Reply:
Definitely a valid question.  I think the initial reasoning was to abstract the `vapid` naming from the client interface, since it isn't a common term.  Now that we have some API versioning in place, we'll need to have the client define which version they want to use anyway, so this feels like the appropriate route.

As far as passing in the base path when making individual requests, that shouldn't be necessary once the `Procore::Client` is initialized.  Borrowing from one of our examples:

```ruby
store = Procore::Auth::Stores::Session.new(session: session)
client = Procore::Client.new(
  client_id: "client id",
  client_secret: "client secret",
  store: store
)

# Get the current user's companies
companies = client.get("companies")
```

Once the `Procore::Client` is initialized, we only have to ask for the specific endpoint name, `companies`, which absolves the user of needing to specify the entire path for each request.

Reply:
Ok, thanks for the explanation.  Personally, if i pass part of the path, I don't mind passing the full path, but I can see what you mean.  I'll make the change. Thanks for reviewing.

Reply:
Thanks for making those changes!